src/apps/activities/services/activity.py

```diff
@@ -256,39 +259,54 @@ async def get_single_language_by_applet_id(
-                is_performance_task=schema.is_performance_task,
-            )
+        try: 
+            schemas = await ActivitiesCRUD(self.session).get_mobile_with_items_by_applet_id(applet_id, is_reviewable=False)
+            activities = []
+            periodicities = []
+            activity_ids = []
+
+            for schema in schemas:
+                activity = ActivityLanguageWithItemsMobileDetailPublic(
+                    id=schema.id,
+                    name=schema.name,
+                    description=self._get_by_language(schema.description, language),
+                    splash_screen=schema.splash_screen,
+                    image=schema.image,
+                    show_all_at_once=schema.show_all_at_once,
+                    is_skippable=schema.is_skippable,
+                    is_reviewable=schema.is_reviewable,
+                    is_hidden=schema.is_hidden,
+                    response_is_editable=schema.response_is_editable,
+                    order=schema.order,
+                    scores_and_reports=schema.scores_and_reports,
+                    performance_task_type=schema.performance_task_type,
+                    is_performance_task=schema.is_performance_task,
+                )
 
-            activities.append(activity)
-            activity_ids.append(activity.id)
+                activities.append(activity)
+                activity_ids.append(activity.id)
 
-        activity_items_map = await ActivityItemService(self.session).get_single_language_by_activity_ids(
-            activity_ids=activity_ids, language=language
-        )
+            activityEvent = await EventCRUD(self.session).get_all_by_activity_flow_ids(applet_id, activity_ids, True)
 
-        for activity in activities:
-            activity.items = activity_items_map.get(activity.id, [])
+            for event in activityEvent:
+                periodicity: Periodicity = await PeriodicityCRUD(self.session).get_by_id(event.periodicity_id)
+                periodicities.append({**periodicity.dict(),'start_time': event.start_time, 'end_time': event.end_time, 'access_before_schedule': event.access_before_schedule, 'one_time_completion': event.one_time_completion, 'timer': event.timer, 'timer_type': event.timer_type, 'event_id': event.id, 'activity_id': event.activity_id })
 
-        return activities
+            activity_items_map = await ActivityItemService(self.session).get_single_language_by_activity_ids(
+                activity_ids=activity_ids, language=language
+            )
+            
+            for activity in activities:
+                for periodicity in periodicities:
+                    if activity.id == periodicity['activity_id']:
+                        activity.periodicity = periodicity
+                        break
+                activity.items = activity_items_map.get(activity.id, [])
+
+            return activities
+
+        except Exception as e:
+            logger.error(e)
```

Comment:
Please, do not use try except only for logging this is hidden some possible errors